src/main.rs

```diff
@@ -737,6 +766,48 @@ fn backtrace(
             false
         };
 
+        // This is our first run through the loop, some initial handling and printing is required
+        // TODO refactor this
+        if frame_index == 0 {
+            if pc & !THUMB_BIT == vector_table.hard_fault & !THUMB_BIT {
+                // HardFaultTrampoline
+                // on hard fault exception entry we hit the breakpoint before the subroutine prelude (`push
+                // lr`) is executed so special handling is required
+                // also note that hard fault will always be the first frame we unwind
+
+                print_backtrace_start();
+
+                let mut stack_overflow = false;
+
+                if let Some(sp_ram_region) = sp_ram_region {
+                    // NOTE stack is full descending; meaning the stack pointer can be
+                    // `ORIGIN(RAM) + LENGTH(RAM)`
+                    let range = sp_ram_region.range.start..=sp_ram_region.range.end;
+                    stack_overflow = !range.contains(&sp);
+
+                    // if a stack overflow happened, we're definitely printing a backtrace
+                    print_backtrace = print_backtrace || stack_overflow;
```

Comment:
could do `print_backtrace |= stack_overflow;` but that's a matter of taste

Reply:
agreed and fixed